test/hotspot/gtest/runtime/test_os_windows.cpp

```diff
@@ -722,6 +722,105 @@ TEST_VM(os_windows, processor_count) {
   }
 }
 
+TEST_VM(os_windows, large_page_init_multiple_sizes) {
+  // Enable large pages and set conditions for the test
+  UseLargePages = true;
+  bool isSupportedWindowsVersion = os::win32::is_windows_11_or_greater() ||
+    os::win32::is_windows_server_2022_or_greater();
+  EnableAllLargePageSizesForWindows = true; // This flag enables support for all large page sizes
+
+  // Determine the minimum page size
+  const size_t min_size = GetLargePageMinimum();
+
+  // Set LargePageSizeInBytes to 4 times the minimum page size
+  LargePageSizeInBytes = 4 * min_size; // Set a value for multiple page sizes
+
+  // Initialize large page settings
+  os::large_page_init();
+
+  // Verify that large pages are enabled
+  if (isSupportedWindowsVersion) {
+    EXPECT_TRUE(UseLargePages) << "UseLargePages should be true after initialization for LargePageSizeInBytes = 4 * min_size";
+  }
+
+  // Verify that _large_page_size is greater than the default page size
+  const size_t default_page_size = os::vm_page_size();
+  size_t _large_page_size = os::large_page_init_decide_size();
+  if (isSupportedWindowsVersion) {
+    EXPECT_GT(_large_page_size, default_page_size) << "Large page size should be greater than the default page size for LargePageSizeInBytes = 4 * min_size";
+  }
+
+#if !defined(IA32)
+  if (isSupportedWindowsVersion && EnableAllLargePageSizesForWindows) {
+    size_t page_size_count = 0;
+    size_t page_size = os::page_sizes().largest();
+
+    do {
+      ++page_size_count;
+      page_size = os::page_sizes().next_smaller(page_size);
+    } while (page_size >= os::page_sizes().smallest());
+
+    EXPECT_GT(page_size_count, 1u) << "There should be multiple large page sizes available.";
+
+    size_t large_page_size = _large_page_size;
+    size_t min_size = os::page_sizes().smallest();
+
+    for (size_t page_size = os::page_sizes().largest(); page_size >= min_size; page_size = os::page_sizes().next_smaller(page_size)) {
+      EXPECT_TRUE(page_size % min_size == 0) << "Each page size should be a multiple of the minimum large page size.";
```

Comment:
Doesn't os::page_sizes().smallest() contain the 4K page size? If so then the message should probably end with "minimum supported page size"

Reply:
> Doesn't os::page_sizes().smallest() contain the 4K page size? If so then the message should probably end with "minimum supported page size"

Indeed, the test needs revision because the page sizes should be multiples of the minimum large page size, which is 2MB, not the minimum page size of 4KB. Essentially, if the list contains more than one entry, each size should be a multiple of the second smallest size.